lib/instrumentation/openai.js

```diff
@@ -173,19 +184,25 @@ module.exports = function initialize(agent, openai, moduleName, shim) {
         promise: true,
         // eslint-disable-next-line max-params
         after(_shim, _fn, _name, err, response, segment) {
+          if (!response) {
+            // If we get an error, it is possible that `response = null`.
+            // In that case, we define it to be an empty object.
+            response = {}
+          }
           response.headers = segment[openAiHeaders]
           response.api_key = segment[openAiApiKey]
 
-          // TODO: add LlmErrorMessage on failure
-          // and exit
-          // See: https://github.com/newrelic/node-newrelic/issues/1845
-          // if (err) {}
-
-          recordChatCompletionMessages({
+          const summary = recordChatCompletionMessages({
             segment,
             request,
             response
           })
+
+          if (err) {
```

Comment:
also if errored we need to add `error: true` to the embedding and summary events